src/main/java/io/gravitee/policy/javascript/JavascriptPolicy.java

```diff
@@ -257,6 +257,9 @@ private String executeStreamScript(JsRequest request, JsResponse response, JsExe
     }
 
     private String eval(String script, ScriptContext scriptContext) throws ScriptException, ExecutionException, InterruptedException {
+        // As per https://github.com/javadelight/delight-nashorn-sandbox/issues/73
+        JAVASCRIPT_ENGINE.eval("Object.defineProperty(this, 'engine', {});\n" + "Object.defineProperty(this, 'context', {});");
+        JAVASCRIPT_ENGINE.eval("delete this.__noSuchProperty__;");
```

Comment:
Can be moved into the JavascriptInitializer to avoid making this eval on each call.